forgerock-openbanking-cdr-aspsp-as/forgerock-openbanking-cdr-aspsp-as-gateway/forgerock-openbanking-cdr-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/TppRegistrationService.java

```diff
@@ -167,19 +169,31 @@ public void verifyTPPRegistrationRequestAgainstSSA(OIDCRegistrationRequest oidcR
     public List<String> parseContacts(JWTClaimsSet ssaClaims) {
         List<String> contacts = new ArrayList<>();
         JSONArray contactsJsonArray = (JSONArray) ssaClaims.getClaim(OpenBankingConstants.SSAClaims.ORG_CONTACTS);
+
         if (contactsJsonArray != null) {
             for (Object contactJson : contactsJsonArray) {
                 JSONObject contactJsonObject = ((JSONObject) contactJson);
                 StringBuilder contact = new StringBuilder();
-                contact.append("email:").append(contactJsonObject.getAsString("email")).append(";");
-                contact.append("name:").append(contactJsonObject.getAsString("name")).append(";");
-                contact.append("phone:").append(contactJsonObject.getAsString("phone")).append(";");
-                contact.append("type:").append(contactJsonObject.getAsString("type")).append(";");
+                contact.append("email:").append(getContactField(contactJsonObject, "email")).append(";");
+                contact.append("name:").append(getContactField(contactJsonObject, "name")).append(";");
+                contact.append("phone:").append(getContactField(contactJsonObject, "phone")).append(";");
+                contact.append("type:").append(getContactField(contactJsonObject, "type")).append(";");
                 contacts.add(contact.toString());
             }
         }
         return contacts;
     }
+
+    private String getContactField(JSONObject contactJsonObject, String field){
+        String fieldValue = null;
+        try{
+            fieldValue = JSONObjectUtils.getString(contactJsonObject, field);
+        } catch(ParseException pe) {
+            log.debug("Warning: ParseException getting field '$1' as string from '$2'",
```

Comment:
Is debug right here?

Reply:
Hmmm. Yeah, maybe a warning would be better?